extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchMergeAggregatorFactory.java

```diff
@@ -142,6 +151,22 @@ public VectorAggregator factorizeVector(VectorColumnSelectorFactory selectorFact
     );
   }
 
+  private void validateInputs(@Nullable ColumnCapabilities capabilities)
+  {
+    if (capabilities != null) {
+      final ColumnType type = capabilities.toColumnType();
+      if (!ColumnType.UNKNOWN_COMPLEX.equals(type) && !TYPE.equals(type) &&
+          !(ValueType.COMPLEX.equals(type.getType()) &&
+            (Objects.equals(type.getComplexTypeName(), "HLLSketch") ||
+             Objects.equals(type.getComplexTypeName(), "HLLSketchBuild")))) {
+        throw new UOE("Using aggregation type %s is not supported for %s column. "
```

Comment:
We don't use UOE nowadays. DruidException is used for better error messages. Let's use that instead.

Reply:
Please format the message according to [style guide](https://github.com/apache/druid/blob/master/dev/style-conventions.md#message-formatting-for-logs-and-exceptions). For example, converting the error message as is means:
```suggestion
        throw new UOE("aggregation type [%s] is not supported for complex columns with type [%s] "
```

Reply:
There are a couple of things that should be changed regarding the error message:
1. "aggregatorType" isn't a thing afaik. We should simply use aggregator
2. The message isn't actionable enough - "Use a different aggregator type and run the query again". What different aggregators can be used by the user? If we cannot glean it from the given type, then let's just remove this portion of the message.